src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/shattered_hallsScripts.cpp

```diff
@@ -270,6 +269,251 @@ struct npc_shattered_hand_centurion : public CombatAI
     }
 };
 
+/*######
+ ## npc_shattered_hand_legionnaire 16700
+ ######*/
+
+enum ShatteredHandLegionnairActions
+{
+    LEGIONNAIRE_PUMMEL,
+    LEGIONNAIRE_AURA_OF_DISCIPLIN,
+    LEGIONNAIRE_ACTION_MAX,
+    LEGIONNAIRE_CALL_FOR_REINFORCEMENTS,
+    LEGIONNAIRE_REINF_CD,
+};
+
+enum ShatteredHandLegionnair
+{
+    SPELL_AURA_OF_DISCIPLINE        = 30472,
+    SPELL_PUMMEL                    = 15615,
+    SPELL_ENRAGE                    = 30485,
+    EMOTE_ENRAGE                    = 1151,
+    MOB_FEL_ORC                     = 17083,   
+};
+
+static float FelOrcSpawnCoords[][4] =               // Coords needed for spawns 
+{
+    { 0.0f, 0.0f, 0.0f, 0.0f},                      // Legionnaire 001 spawn coords
+    { 79.9949f, 111.5607f, -13.1384f, 4.6949f},     // Legionnaire 002 right side felorc spawn
+    { 61.1264f, 110.8250f, -13.1384f, 6.1784f },    // Legionnaire 002 left side felorc spawn
+    { 88.4735f, 187.3315f, -13.1929f, 3.144f },     // Legionnaire 003 spawn
+    { 78.6885f, 218.2196f, -13.2166f, 4.013f },     // Legionnaire 004 spawn
+    { 83.5307f, 250.5344f, -13.1131f, 3.060f },     // Legionnaire 005 spawn
+    { 69.8188f, 239.513f, -13.193643f, 0.0 }         // Legionnaire 006 waypoint
+};
+
+static const int32 aRandomAggro[] = { 16700, 16703, 16698, 16701, 16702, 16697, 16699 };
+
+static const int32 aRandomReinf[] = { 16356, 16357, 16358, 16359, 16360, 16361, 16362 };
+static const int32 aRandomReinfSleeping[] = { 16363, 16364, 16365, 16366, 16367 };
+
+struct npc_shattered_hand_legionnaire : public CombatAI
+{
+    npc_shattered_hand_legionnaire(Creature* creature) : CombatAI(creature, LEGIONNAIRE_ACTION_MAX), m_instance(static_cast<instance_shattered_halls*>(creature->GetInstanceData())),
+    m_reinfCD(false)
+    {
+        AddCombatAction(LEGIONNAIRE_PUMMEL, 10000, 15000);
+        AddCombatAction(LEGIONNAIRE_AURA_OF_DISCIPLIN, 0, 5000);
+        AddCustomAction(LEGIONNAIRE_CALL_FOR_REINFORCEMENTS, true, [&]() { CallForReinforcements(); });
```

Comment:
Maybe mark it with param TIMER_COMBAT_COMBAT?, not sure if it can occur out of combat.

Reply:
Reinforcement calls can happen out of combat too, rly important for the big group at the start:
https://youtu.be/gtUP0mA6V7E